crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -18,83 +18,131 @@ pub(super) fn place_comment<'a>(
+            | AnyNodeRef::ExprDictComp(_)
+            | AnyNodeRef::ExprTuple(_) => handle_bracketed_end_of_line_comment(comment, locator),
+            _ => CommentPlacement::Default(comment),
+        })
+}
+
+fn handle_parenthesized_comment<'a>(
+    comment: DecoratedComment<'a>,
+    locator: &Locator,
+) -> CommentPlacement<'a> {
+    let Some(preceding) = comment.preceding_node() else {
+        return CommentPlacement::Default(comment);
     };
 
-    // Change comment placement depending on the node type. These can be seen as node-specific
-    // fixups.
-    comment.or_else(|comment| match comment.enclosing_node() {
-        AnyNodeRef::Parameters(arguments) => {
-            handle_parameters_separator_comment(comment, arguments, locator)
-                .or_else(|comment| handle_bracketed_end_of_line_comment(comment, locator))
-        }
-        AnyNodeRef::Arguments(_) | AnyNodeRef::TypeParams(_) => {
-            handle_bracketed_end_of_line_comment(comment, locator)
-        }
-        AnyNodeRef::Comprehension(comprehension) => {
-            handle_comprehension_comment(comment, comprehension, locator)
-        }
-        AnyNodeRef::ExprAttribute(attribute) => handle_attribute_comment(comment, attribute),
-        AnyNodeRef::ExprBinOp(binary_expression) => {
-            handle_trailing_binary_expression_left_or_operator_comment(
-                comment,
-                binary_expression,
-                locator,
-            )
-        }
-        AnyNodeRef::Keyword(_) => handle_dict_unpacking_comment(comment, locator),
-        AnyNodeRef::ExprDict(_) => handle_dict_unpacking_comment(comment, locator)
-            .or_else(|comment| handle_bracketed_end_of_line_comment(comment, locator)),
-        AnyNodeRef::ExprIfExp(expr_if) => handle_expr_if_comment(comment, expr_if, locator),
-        AnyNodeRef::ExprSlice(expr_slice) => handle_slice_comments(comment, expr_slice, locator),
-        AnyNodeRef::ExprStarred(starred) => {
-            handle_trailing_expression_starred_star_end_of_line_comment(comment, starred)
-        }
-        AnyNodeRef::ExprSubscript(expr_subscript) => {
-            if let Expr::Slice(expr_slice) = expr_subscript.slice.as_ref() {
-                handle_slice_comments(comment, expr_slice, locator)
-            } else {
-                CommentPlacement::Default(comment)
+    let Some(following) = comment.following_node() else {
+        return CommentPlacement::Default(comment);
+    };
+
+    // This will need to support a full grammar between expressions (so `and`, `or`, `%`, etc.).
+    // It would be nice if we could just store the locations of all open parentheses.
+    if comment.line_position().is_end_of_line() {
+        for token in SimpleTokenizer::new(
+            locator.contents(),
+            TextRange::new(preceding.end(), comment.start()),
+        )
+        .skip_trivia()
+        .filter(|token| token.kind() == SimpleTokenKind::LParen)
```

Comment:
Can we limit the non-trivia tokens that we are comfortable skipping? I would assume that we only ever have to skip LParens, or are there other tokens?